### PR TITLE
[AzureMonitorExporter][AzureMonitorDistro] update otel packages to 1.7.0

### PR DESCRIPTION
**Pull request by TimothyMothra**
Supersedes #41370

### Changes
- Update all OTel dependencies to v1.7.0
- Add stable release of OpenTelemetry.Instrumentation.AspNetCore to our Distro
- Add stable release of OpenTelemetry.Instrumentation.Http to our Distro
- Update `OpenTelemetryBuilderExtensions.cs` to add Instrumentations to TracerProviderBuilder.
  - this includes customizations for Azure SDK
- Remove unused package references.

### Misc Changes
- fix nullable in LogsHelper.cs (Exporter project)
- Detected package downgrade: Microsoft.Extensions.Logging from 8.0.0 (Exporter test projects)
  - fix is to remove explicit reference to either 6.0.0 or 8.0.0-rc from our projects
- Detected package downgrade: Microsoft.Bcl.AsyncInterfaces from 8.0.0 to 7.0.0 (Distro test projects)
  - fix is to upgrade explicit reference from 7.0.0 to 8.0.0
- Found conflicts between different versions of "Microsoft.Bcl.AsyncInterfaces" that could not be resolved (LiveMetrics test projects)
  - fix is to add an explicit reference to 8.0.0

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/OpenTelemetryBuilderExtensions.cs [hunk 1]
```diff
@@ -96,6 +96,8 @@ public static OpenTelemetryBuilder UseAzureMonitor(this OpenTelemetryBuilder bui
             builder.WithTracing(b => b
                             .AddSource("Azure.*")
                             .AddVendorInstrumentationIfPackageNotReferenced()
+                            .AddAspNetCoreInstrumentation()
+                            .AddHttpClientInstrumentation()
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/OpenTelemetryBuilderExtensions.cs:100]: Missing customizations for Azure SDK.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Azure.Monitor.OpenTelemetry.AspNetCore.csproj [hunk 2]
```diff
@@ -14,10 +14,12 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="OpenTelemetry.Extensions.Hosting" />
     <PackageReference Include="System.Text.Encodings.Web" />
     <PackageReference Include="Microsoft.AspNetCore.Http.Abstractions" />
     <PackageReference Include="Microsoft.AspNetCore.Http.Features" />
```
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Azure.Monitor.OpenTelemetry.AspNetCore.csproj:19]: Do we still need these references `Microsoft.AspNetCore.Http.Abstractions`, `System.Text.Encodings.Web` and `Microsoft.AspNetCore.Http.Features`?

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/OpenTelemetryBuilderExtensions.cs (hunk 1 above)
Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/OpenTelemetryBuilderExtensions.cs]: done. please re-review.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Azure.Monitor.OpenTelemetry.AspNetCore.csproj (hunk 2 above)
Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Azure.Monitor.OpenTelemetry.AspNetCore.csproj:19]: I don't know the history of each packagereference. looks like there are some in both projects that could be removed. I'll do that now.

**Review by rajkumar-rangaraj — APPROVED** [review]

**Review by vishweshbankwar — APPROVED** [review]
We need to update the Readme as well (separate PR)

**Pull request merged**